src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -57,7 +59,7 @@
  *
  * @author Kohsuke Kawaguchi
  */
-public class GitHubPushTrigger extends Trigger<Job<?, ?>> implements GitHubTrigger {
+public class GitHubPushTrigger extends Trigger<Job<?, ?>> implements GitHubTrigger2 {
```

Comment:
so we can just continue to implement old interface (marked as deprecated) and do nothing with its methods and just use brand new method